packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,395 @@
+    if (_previousValue == value) {
+      return;
+    }
+    _previousValue = value;
+    notifyListeners();
+  }
+
+  OutlinedBorder get shape => _shape!;
+  OutlinedBorder? _shape;
+  set shape(OutlinedBorder value) {
+    if (_shape == value) {
+      return;
+    }
+    _shape = value;
+    notifyListeners();
+  }
+
+  BorderSide? get side => _side;
+  BorderSide? _side;
+  set side(BorderSide? value) {
+    if (_side == value) {
+      return;
+    }
+    _side = value;
+    notifyListeners();
+  }
+
+  Rect _outerRectAt(Offset origin) {
+    const double size = CupertinoCheckbox.width;
+    final Rect rect = Rect.fromLTWH(origin.dx, origin.dy, size, size);
+    return rect;
+  }
+
+  // The checkbox's border color if value == false, or its fill color when
+  // value == true or null.
+  Color _colorAt(bool value) {
+    return value && isActive ? activeColor : inactiveColor;
+  }
+
+  // White stroke used to paint the check and dash.
+  Paint _createStrokePaint() {
+    return Paint()
+      ..color = checkColor
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2.5
+      ..strokeCap = StrokeCap.round;
+  }
+
+  void _drawBox(Canvas canvas, Rect outer, Paint paint, BorderSide? side, bool fill) {
+    if (fill) {
+      canvas.drawPath(shape.getOuterPath(outer), paint);
+    }
+    if (side != null) {
+      shape.copyWith(side: side).paint(canvas, outer);
+    }
+  }
+
+  void _drawCheck(Canvas canvas, Offset origin, Paint paint) {
+    final Path path = Path();
+    // The ratios for the offsets below were found from looking at the checkbox
+    // examples on in the HIG docs. The distance from the needed point to the
+    // edge was measured, then devided by the total width.
+    const Offset start = Offset(CupertinoCheckbox.width * 0.25, CupertinoCheckbox.width * 0.52);
+    const Offset mid = Offset(CupertinoCheckbox.width * 0.46, CupertinoCheckbox.width * 0.75);
+    const Offset end = Offset(CupertinoCheckbox.width * 0.72, CupertinoCheckbox.width * 0.29);
+    path.moveTo(origin.dx + start.dx, origin.dy + start.dy);
+    path.lineTo(origin.dx + mid.dx, origin.dy + mid.dy);
+    canvas.drawPath(path, paint);
+    path.moveTo(origin.dx + mid.dx, origin.dy + mid.dy);
+    path.lineTo(origin.dx + end.dx, origin.dy + end.dy);
+    canvas.drawPath(path, paint);
+  }
+
+  void _drawDash(Canvas canvas, Offset origin, Paint paint) {
+    // From measuring the checkbox example in the HIG docs, the dash was found
+    // to be half the total width, centered in the middle.
+    const Offset start = Offset(CupertinoCheckbox.width * 0.25, CupertinoCheckbox.width * 0.5);
+    const Offset end = Offset(CupertinoCheckbox.width * 0.75, CupertinoCheckbox.width * 0.5);
+    canvas.drawLine(origin + start, origin + end, paint);
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Paint strokePaint = _createStrokePaint();
+    final Offset origin = size / 2.0 - const Size.square(CupertinoCheckbox.width) / 2.0 as Offset;
```

Comment:
Too bad we have to cast here (but we do). Someday we should add a `asOffset` to `Size` and a `asSize` to `Offset`.